File: lua/2.lua
--project euler 2: find the sum of all even Fibonacci numbers below 4,000,000
package.path = "..\\libraries\\?.lua;" .. package.path
require "lib"
local t = os.clock()

local sum = 0

for n in lib.fibonacci(4e6) do
	if (n%2 == 0) then
		sum = sum+n
	end
end

print("result: "..sum.." - found in "..(os.clock() - t).." s")

File: python/2.py
# Project Euler 2: find the sum of all even Fibonacci numbers below 4,000,000

#solution 1
sum = 0
a, b = 1, 1
while a < 4000000:
	b, a = a+b, b
	if a%2 == 0:
		sum += a

print(sum)

#solution 2
fib = [1,1]
while fib[-1] < 4e6:
	fib.append(fib[-2] + fib[-1])

print(sum([x for x in fib if x % 2 == 0]))